src/extension/commandPaletteHandler.ts

```diff
@@ -175,18 +192,21 @@ export class CommandPaletteHandler {
     public static runExponent(): Q.Promise<void> {
         return this.selectProject()
             .then((project: IReactNativeProject) => {
-                return this.loginToExponent(project)
-                    .then(() => {
-                        return this.executeCommandInContext("runExponent", project.workspaceFolder, () => {
-                            const platform = <ExponentPlatform>this.createPlatform(project, "exponent", ExponentPlatform);
-                            return platform.beforeStartPackager()
-                                .then(() => {
-                                    return platform.startPackager();
-                                })
-                                .then(() => {
-                                    return platform.runApp();
+                return CommandPaletteHandler.checkReactNativePackageExistence(project.workspaceFolder.uri.path)
+                    .then(version => {
+                        return this.loginToExponent(project)
+                            .then(() => {
+                                return this.executeCommandInContext("runExponent", project.workspaceFolder, () => {
+                                    const platform = <ExponentPlatform>this.createPlatform(project, "exponent", ExponentPlatform);
+                                    return platform.beforeStartPackager()
+                                        .then(() => {
+                                            return platform.startPackager();
+                                        })
+                                        .then(() => {
+                                            return platform.runApp();
+                                        });
                                 });
-                        });
+                            });
```

Comment:
Indentation here

Reply:
Could you please clarify what indentation is incorrect?

Reply:
Hm, my mistake.